crates/ruff/tests/format.rs

```diff
@@ -90,6 +90,172 @@ fn format_warn_stdin_filename_with_files() {
     "###);
 }
 
+#[test]
+fn nonexistent_config_file() {
+    assert_cmd_snapshot!(Command::new(get_cargo_bin(BIN_NAME))
+        .args(["format", "--config", "foo.toml", "."]), @r###"
+        success: false
+        exit_code: 2
+        ----- stdout -----
+
+        ----- stderr -----
+        error: invalid value 'foo.toml' for '--config <CONFIG_OPTION>'
+
+          tip: The `--config` flag must either be a path to a `.toml` configuration file or a TOML `<KEY> = <VALUE>` pair overriding a specific config setting
+          tip: The path `foo.toml` does not exist on your filesystem
+
+        For more information, try '--help'.
+       "###);
+}
+
+#[test]
+fn config_override_rejected_if_invalid_toml() {
+    assert_cmd_snapshot!(Command::new(get_cargo_bin(BIN_NAME))
+        .args(["format", "--config", "foo = bar", "."]), @r###"
+    success: false
+    exit_code: 2
+    ----- stdout -----
+
+    ----- stderr -----
+    error: invalid value 'foo = bar' for '--config <CONFIG_OPTION>'
+
+      tip: The `--config` flag must either be a path to a `.toml` configuration file or a TOML `<KEY> = <VALUE>` pair overriding a specific config setting
+      tip: The following error occurred when attempting to parse `foo = bar` as a `ruff.toml` config option:
+
+    TOML parse error at line 1, column 7
+      |
+    1 | foo = bar
+      |       ^
+    invalid string
+    expected `"`, `'`
+
+    For more information, try '--help'.
+    "###);
+}
+
+#[test]
+fn too_many_config_files() -> Result<()> {
+    let tempdir = TempDir::new()?;
+    let ruff_dot_toml = tempdir.path().join("ruff.toml");
+    let ruff2_dot_toml = tempdir.path().join("ruff2.toml");
+    fs::File::create(&ruff_dot_toml)?;
+    fs::File::create(&ruff2_dot_toml)?;
+    let expected_stderr = format!(
+        "\
+ruff failed
+  Cause: You cannot specify more than one configuration file on the command line.
+
+  tip: remove either `--config={}` or `--config={}`.
+       For more information, try `--help`.
+
+",
+        ruff_dot_toml.display(),
+        ruff2_dot_toml.display(),
+    );
+    let cmd = Command::new(get_cargo_bin(BIN_NAME))
+        .arg("format")
+        .arg("--config")
+        .arg(&ruff_dot_toml)
+        .arg("--config")
+        .arg(&ruff2_dot_toml)
+        .arg(".")
+        .output()?;
+    let stderr = std::str::from_utf8(&cmd.stderr)?;
+    assert_eq!(stderr, expected_stderr);
+    Ok(())
+}
+
+#[test]
+fn config_file_and_isolated() -> Result<()> {
+    let tempdir = TempDir::new()?;
+    let ruff_dot_toml = tempdir.path().join("ruff.toml");
+    fs::File::create(&ruff_dot_toml)?;
+    let expected_stderr = format!(
+        "\
+ruff failed
+  Cause: The argument `--config={}` cannot be used with `--isolated`
```

Comment:
I would omit the `={}` here if we retain this error and write as "The `--config` argument cannot be used with `--isolated`"

Reply:
Hmm -- _some_ `--config` arguments are allowed, though. It's okay to use the `--config` flag alongside `--isolated` if `--config` is being used to override a specific configuration option. It's just not okay to use `--config` alongside `--isolated` if `--config` is being used to provide a configuration file. So the value being passed to `--config` does matter here